samples/openthread/cli/README.rst

```diff
@@ -64,6 +64,20 @@ The provided configurations optimize the memory footprint of the sample for sing
 
 For more information, see :ref:`app_memory`.
 
+.. ot_cli_sample_fem_support_start
+
+FEM support
+===========
+
+Support for nRF21540 front-end module as specified in :ref:`ug_radio_fem` can be added to the Thread samples by using the provided devicetree overlay file.
+
+For example, for an nRF52833 DK with an attached nRF21540 EK, invoke the following command within the sample directory:
+``west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="boards/nrf52833dk_nrf52833.overlay;../common/dts-nrf21540-fem.overlay"``
+
+A similar approach can be used for adding support to other front-end modules.
```

Comment:
```suggestion
You can add support for the nRF21540 front-end module to the sample.
See :ref:`ug_radio_fem` for more information.

To add support for the nRF21540 FEM, add the provided :file:`dts-nrf21540-fem.overlay` devicetree overlay file when building.
The file is located in the :file:`samples/openthread/common` folder.
It must be added to the ``DTC_OVERLAY_FILE`` variable (see :ref:`zephyr:important-build-vars`).
Follow the instructions in :ref:`cmake_options` to add this option.

.. note::
   You must add the provided overlay file if you use the nRF21540 EK.
   If you use the nRF21540 DK, build your application for the :ref:`nrf21540dk_nrf52840 <zephyr:nrf21540dk_nrf52840>` board.
   The devicetree for the nRF21540 DK already contains the required FEM configuration, so you do not need to add the overlay file.

For example, to build the sample from the command line for an nRF52833 DK with an attached nRF21540 EK, invoke the following command within the sample directory:
``west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="boards/nrf52833dk_nrf52833.overlay;../common/dts-nrf21540-fem.overlay"``

To add support for other front-end modules, add the respective overlay files in the same way.
```

Reply:
Is the note about the EK vs. DK correct?
PR #3579 adds support for the DK, so I guess you only need this extra step for the EK.

Reply:
Unrelated to this PR, but something we might need to look into: Why do we need the nRF52833 DK overlay file? I can't find any information about this in the documentation.

Reply:
Good question. I think, that overlays named same as the board should be applied automatically, but maybe when we specify additional dts overlays this does not work as expected. @edmont Could you confirm this is needed?

Reply:
sry, wrong account:)

Reply:
Yes, the problem is that if we do something like this:

`west build -b nrf52833dk_nrf52833 -- -DDTC_OVERLAY_FILE="../common/dts-nrf21540-fem.overlay"`

Then the file `boards/nrf52833dk_nrf52833.overlay` won't be loaded. I didn't find an easier way to overcome this in the CMake documentation.

Reply:
So if you build without the EK, you would just use "west build -b nrf52833dk_nrf52833"? But with the EK, you need to specify both overlay files?
This seems very confusing, and users will need to know about it. Is that something specific to Thread? Or a general issue?
And how does it work in SES?

Reply:
I agree it's confusing, that's why an example is given here. For Kconfig files we have `OVERLAY_CONFIG` which doesn't modify the `CONF_FILE`, but for the case of Device Tree `DTC_OVERLAY_FILE` takes precedence over the board DTC file.

At the end it's a matter of combining different DTC files: Zephyr board defaults, sample specifics and feature specifics. Another option would be to add the FEM definitions to just one sample so that the user understands the concept, but that seems less reusable. Maybe a better solution would be to modify the CMake preprocessor to always load the sample specific DTC file regardless of the `DTC_OVERLAY_FILE` value, but that's out of scope of this PR.

Reply:
I'm concerned that the example is more confusing than helpful. It might help to be more specific and explain that while you normally don't specify the board overlay file, you need to do so when you specify a second overlay file.

The example it's only for command-line builds and not SES - does the same apply when building in SES?
Also, the example is only for one board, but I guess it's the same for nRF52840. I don't see nRF5340 in the boards folder - is that one different and you only specify the FEM overlay file in that case?

Reply:
@ru-fu I made some changes for more clarification. As for SES, I'm not able to test it at the moment, but there should be enough information on how to configure it with this option.

In any case, I believe @MarekPorwisz mentioned about the need of going all over the Thread samples and make sure clear information about using SES is provided.